internal/transform/transform_test.go

```diff
@@ -252,10 +252,15 @@ func TestFullTransform(t *testing.T) {
 			want: `<Component><h1>Hello world</h1></Component>`,
 		},
 		{
-			name:   "Component before html",
+			name:   "Component before html I",
 			source: `<Navigation /><html><body><h1>Astro</h1></body></html>`,
 			want:   `<Navigation></Navigation><h1>Astro</h1>`,
 		},
+		{
+			name:   "Component before html II",
+			source: `<MainHead title={title} description={description} /><html lang="en"><body><slot /></body></html>`,
+			want:   `<MainHead title={title} description={description}></MainHead><slot></slot>`,
```

Comment:
The output looks incorrect but I believe it's intended?
(though this isn't related to this PR or #891. I'll open an issue so we can discuss about it)

Reply:
Hmm, both of these seem incorrect, to be honest. I'd expect the `<html>` and `<body>` to be preserved.

Reply:
Could we maybe merge this for now and close the issue as it wasn't actually about the incorrect markup, but the `index out of range` errors which have been fixed?

This bug apparently existed since at least v1.4.3 (which is the oldest version my playground supports) for normal elements.

I opened a new issue for this specific bug https://github.com/withastro/compiler/issues/913

Reply:
Yep that seems reasonable!